tools/migration/internal/runner.go

```diff
@@ -27,22 +40,112 @@ type Migration interface {
 }
 
 type MigrationRunner struct {
-	verbose    bool
-	command    string
+	// verbose controls the amount of output.
+	verbose bool
+
+	// command is the migration command to run, passed from the CLI
+	command string
+
+	// oldRepoOpt is the option passed  from the CLI.
 	oldRepoOpt string
+
+	// MigrationsProvider is a dependency for fetching available migrations
+	// to allow unit tests to supply test migrations without creating test fixtures.
+	MigrationsProvider func() []Migration
+
+	// RepoVersionGetter is a dependency for getting a Repo Version,
+	// to allow the binary version of the repo to be mocked in testing.
+	RepoVersionGetter func() uint
```

Comment:
I see this is intended to abstract the hardcoded `repo.Version` constant. I don't think migration code should be referring to that constant at all. The version of an existing repo should be determined by inspecting it. That constant will need factoring out from almost all the places that refer to it,  its only use being to check that a repo loaded by go-filecoin matches the expected version.